test/NSHM_NZ/inversion/BuildCrustalFaultSubsectionsTest.java

```diff
@@ -0,0 +1,136 @@
+
+import static org.junit.Assert.*;
+
+import com.google.common.base.Joiner;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.InputStream;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import org.dom4j.Document;
+import org.dom4j.DocumentException;
+import org.dom4j.DocumentException;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.opensha.commons.data.CSVFile;
+import org.opensha.commons.geo.Location;
+import org.opensha.commons.util.IDPairing;
+import org.opensha.commons.util.XMLUtils;
+import org.opensha.refFaultParamDb.vo.FaultSectionPrefData;
+import org.opensha.sha.faultSurface.FaultSection;
+import org.opensha.sha.faultSurface.FaultTrace;
+
+import scratch.UCERF3.enumTreeBranches.DeformationModels;
+import scratch.UCERF3.enumTreeBranches.FaultModels;
+import scratch.UCERF3.enumTreeBranches.ScalingRelationships;
+import scratch.UCERF3.enumTreeBranches.SlipAlongRuptureModels;
+import scratch.UCERF3.inversion.coulomb.CoulombRates;
+import scratch.UCERF3.inversion.InversionFaultSystemRupSet;
+import scratch.UCERF3.inversion.laughTest.UCERF3PlausibilityConfig;
+import scratch.UCERF3.inversion.SectionCluster;
+import scratch.UCERF3.inversion.SectionClusterList;
+import scratch.UCERF3.logicTree.LogicTreeBranch;
+import scratch.UCERF3.utils.DeformationModelFetcher;
+import scratch.UCERF3.utils.FaultSystemIO;
+
+import scratch.UCERF3.inversion.UCERF3SectionConnectionStrategy;
+
+/* 
+ * Objective: Examine how the subsections are build.
+ * 
+ * Fault sections are subdivided such that subsections have length at least(or is it at most?) 0.5 of the seismogenic width - complete down-dip width DDW. 
+ * DDW is not a measure of vertical depth, but of rupture extent that encompasses DDW. Fault-dip angle have an key role.
+ * 
+ * Questions:
+ * (1) Are width of each subsection appropriately computed (whether do 2 subsections ensures L~W). 
+ * (2) How are exceptions (such as fault-sections with length shorter than seismogenic width handled) handled.
+ *          
+ * @author Thingbaijam        
+ * 
+ * Note to myself:
+ * 
+ * getSubSectionsList() is declared (overloaded) in the interface FaultSection and implemented in class FaultSectionPrefData.
+ * Class DeformationModelFetcher also has a method of same name, which ultimately can be connected to the implementation in FaultSectionPrefData.
+ * 
+ *    
+*/
+
+
+public class BuildCrustalFaultSubsectionsTest {
+
+	private static final double grid_disc = 5d;
+
+	
+	static List<FaultSection> fsd;
+	
+	@BeforeClass
+	public static void setUpBeforeClass() throws Exception {
+		// this is the input fault section data file
+		// Still use alderman_sections.xml
+		
+		File fsdFile = new File("./test/NSHM_NZ/inversion/fixtures/alderman_sections.xml");
```

Comment:
NB - I think there's a 'proper` way to include fixtures as java resources. I haven't figured out how this should be done in this project yet. I tried messing with the ClassLoader but with no real success.

Reply:
Okay - that's a good point.